projects/libpng/build_w_pass.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash -eu
+
+# Copyright 2017-2018 Glenn Randers-Pehrson
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Revisions by Glenn Randers-Pehrson, 2017:
+# 1. Build only the library, not the tools (changed "make -j$(nproc) all" to
+#     "make -j$(nproc) libpng16.la").
+# 2. Disabled WARNING and WRITE options in pnglibconf.dfa.
+# 3. Build zlib alongside libpng
+################################################################################
+
+# Disable logging via library build configuration control.
+cat scripts/pnglibconf.dfa | \
+  sed -e "s/option STDIO/option STDIO disabled/" \
+      -e "s/option WARNING /option WARNING disabled/" \
+      -e "s/option WRITE enables WRITE_INT_FUNCTIONS/option WRITE disabled/" \
+> scripts/pnglibconf.dfa.temp
+mv scripts/pnglibconf.dfa.temp scripts/pnglibconf.dfa
+
+# build the libpng library.
+autoreconf -f -i
+./configure --with-libpng-prefix=OSS_FUZZ_
+make -j$(nproc) clean
+make -j$(nproc) libpng16.la
+
+# Add flags
+export REPORT_FLAGS="-Xclang -load -Xclang /src/libpng/ReportFunctionExecutedPass/libReportPass.so -flegacy-pass-manager"
+# Might not need the CFLAGS since it doesn't look like any C files are being compiled
+export CFLAGS="${CFLAGS:=} $REPORT_FLAGS /src/libpng/ReportFunctionExecutedPass/reporter.c++.o -lc++ -pthread -lm"
+export CXXFLAGS="${CXXFLAGS:=} $REPORT_FLAGS /src/libpng/ReportFunctionExecutedPass/reporter.c++.o -pthread -lm"
```

Comment:
the pass and reporter should be added before the project is compiled, 

i.e. before 35
```sh
./configure ...
make -j ...
```